modules/apps/foundation/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/facet/display/context/UserSearchFacetDisplayContext.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.portal.search.web.internal.facet.display.context;
+
+import com.liferay.portal.kernel.search.facet.Facet;
+import com.liferay.portal.kernel.search.facet.collector.FacetCollector;
+import com.liferay.portal.kernel.search.facet.collector.TermCollector;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * @author Lino Alves
+ */
+public class UserSearchFacetDisplayContext {
+
+	public UserSearchFacetDisplayContext(
+		Facet facet, String fieldParam, int frequencyThreshold, int maxTerms,
+		boolean showFrequencies) {
+
+			_facet = facet;
+			_fieldParam = fieldParam;
+			_frequencyThreshold = frequencyThreshold;
+			_maxTerms = maxTerms;
+			_showFrequencies = showFrequencies;
+			_userName = fieldParam;
+		}
+
+		public String getFieldParamInputName() {
+			return _facet.getFieldId();
+		}
+
+		public String getFieldParamInputValue() {
+			return _fieldParam;
+		}
+
+		public List<UserSearchFacetTermDisplayContext>
+		getTermDisplayContexts() {
+
+			FacetCollector facetCollector = _facet.getFacetCollector();
+
+			List<TermCollector> termCollectors =
+				facetCollector.getTermCollectors();
+
+			if (termCollectors.isEmpty()) {
+				return getEmptySearchResultTermDisplayContexts();
+			}
+
+			List<UserSearchFacetTermDisplayContext>
+				userSearchFacetTermDisplayContexts = new ArrayList<>(
+					termCollectors.size());
+
+			int limit = termCollectors.size();
+
+			if ((_maxTerms > 0) && (limit > _maxTerms)) {
+				limit = _maxTerms;
+			}
+
+			for (int i = 0; i < limit; i++) {
```

Comment:
Can we keep the original `for`/`break` logic? (https://github.com/linolaoj/liferay-portal/blob/da6d8d09a949bb4c19d1f95de6dc451db8d389fa/modules/apps/foundation/portal-search/portal-search-web/src/main/resources/META-INF/resources/facets/view/users.jsp#L46-L53) There's a subtle difference regarding `frequencyThreshold` and I'm not sure if it was intentional or not... want to play it safe for now